packages/service-proxy/test/mock-service.connector.ts

```diff
@@ -51,17 +51,24 @@ export class MockConnector {
   }
 
   get DataAccessObject() {
-    return {
-      geocode: async function(street: string, city: string, zipcode: string) {
-        return {
-          lat: 37.5669986,
-          lng: -122.3237495,
-        };
-      },
+    const dao = this.connected
```

Comment:
Hmm, I find this a bit ugly.

IIUC, by returning a different object for connected/not-connected case, you are trying to mimic the behavior of our SOAP connector. Is that right? In that case, please add a comment to explain this to readers.

To keep the code easier to read and this patch smaller, please rework `get DataAccessObject` to use reverse logic and early return.

```ts
get DataAccessObject() {
  if (!this.connected) {
    // (add a comment explaining why we are doing this)
   return {};
  }

  return {
    // the original data-access object
    // (this code block remains unchanged)
  };
}
```